### PR TITLE
HSEARCH-2113 Improve sort support

### PR DESCRIPTION
**Pull request by gsmet**
In passing, create an infrastructure to be able to ignore specific tests which don't make sense with Elasticsearch.

### CODE REVIEW

**Comment by gunnarmorling** [comment]
Nice one. Maybe just a shorter name for the category class, e.g. `SkipOnElasticsearch`?

File: engine/src/test/java/org/hibernate/search/testsupport/junit/TestExcludedWithElasticsearchCategory.java [hunk 1]
```diff
@@ -0,0 +1,11 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.testsupport.junit;
+
+public interface TestExcludedWithElasticsearchCategory {
```
Comment (gunnarmorling) [inline engine/src/test/java/org/hibernate/search/testsupport/junit/TestExcludedWithElasticsearchCategory.java]: A comment would be nice, describing this a JUnit category marker.

**Comment by gsmet** [comment]
I made the necessary changes. I also added an ElasticsearchSupportInProgress category to be able to temporarily exclude tests and and track them afterwards. Might be useful when a test is testing a lot of infrastructure (yes, DSLTest, I'm thinking about you!).

File: elasticsearch/pom.xml [hunk 2]
```diff
@@ -314,6 +313,7 @@
                                 <exclude>**/FSSlaveAndMasterDPTest.java</exclude>
                                 <exclude>**/IndexReaderSeeOptimizedIndexTest.java</exclude>
                             </excludes>
+                            <excludedGroups>org.hibernate.search.testsupport.junit.SkipOnElasticsearch,org.hibernate.search.testsupport.junit.ElasticsearchSupportInProgress</excludedGroups>
```
Comment (Sanne) [inline elasticsearch/pom.xml]: +1 for this approach, but could we have it separated from the "improve Sort support" issue?

Comment (gsmet) [inline elasticsearch/pom.xml]: Working on it.

Comment (Sanne) [inline elasticsearch/pom.xml]: great, thanks!

Comment (gsmet) [inline elasticsearch/pom.xml]: Done. It's now 2 different commits.

**Comment by Sanne** [comment]
merged! Great progress :+1:

**Pull request closed without merging**